identity/ipa/ipa_host.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ipa_host
+short_description: Manage FreeIPA host
+description:
+- Add, modify and delete an IPA host using IPA API
+options:
+  fqdn:
+    description:
+    - Full qualified domain name.
+    - Can not be changed as it is the unique identifier.
+    required: true
+    aliases: ["name"]
+  description:
+    description:
+    - A description of this host.
+    required: false
+  force:
+    description:
+    - Force host name even if not in DNS.
+    required: false
+  ip_address:
+    description:
+    - Add the host to DNS with this IP address.
+    required: false
+  macddress:
```

Comment:
There is no such policy or preference. The only preference would be to keep the same param names for the same subjects, like `force`, `state` and `name`, etc.

if you however like to keep the variables as is, you could also consider aliases in the arg spec.

~~~python
            macaddress=dict(type='list', required=False, aliases=['mac_address']),
~~~

and change in the docs the name and the alias vis versa:

~~~yaml
  mac_address:
    description:
    - List of Hardware MAC address(es) off this host.
    - If option is omitted MAC addresses will not be checked or changed.
    - If an empty list is passed all assigned MAC addresses will be removed.
    - MAC addresses that are already assigned but not passed will be removed.
    required: false
    aliases: [ 'macaddress']
~~~

This is however just a suggestion.